### PR TITLE
Open request component

### PR DESCRIPTION
**Pull request by ryanlieu**
### Summary of Changes

Adds an Open Request Metadata component that triggers the opening/closing of the request component.
### Tests

Need to add tests for the component after implementation is approved

### Documentation

N/A

### CheckList
Make sure you have checked **all** steps below to ensure a timely review.
- [ ] PR title addresses the issue accurately and concisely. Example: "Updates the version of Flask to v1.0.2"
    - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
- [ ] PR includes a summary of changes, including screenshots of any UI changes. 
- [ ] PR adds unit tests, updates existing unit tests, __OR__ documents why no test additions or modifications are needed.
- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
    - All the public python functions and the classes in the PR contain docstrings that explain what it does
- [ ] PR passes all tests documented in the [developer guide](https://github.com/lyft/amundsenfrontendlibrary/blob/master/docs/developer_guide.md#testing)
- [ ] I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"

### CODE REVIEW

File: amundsen_application/static/js/components/TableDetail/OpenRequestMetadata/index.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+interface StateFromProps {
+  requestIsOpen: boolean;
+}
+
+export interface DispatchFromProps {
+  toggleRequest: () => ToggleRequestAction;
+}
+
+export type OpenRequestMetadataProps = StateFromProps & DispatchFromProps;
```
Comment (ttannis) [inline amundsen_application/static/js/components/TableDetail/OpenRequestMetadata/index.tsx]: nit nit: I think "RequestDescription" is a more succinct and appropriate name.

File: amundsen_application/static/js/components/TableDetail/OpenRequestMetadata/index.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+export class OpenRequestMetadata extends React.Component<OpenRequestMetadataProps, OpenRequestMetadataState> {
+  public static defaultProps: Partial<OpenRequestMetadataProps> = {};
+
+  constructor(props) {
+    super(props);
+  }
+
+  openRequest = () => {
+    if (!this.props.requestIsOpen) {
```
Comment (ttannis) [inline amundsen_application/static/js/components/TableDetail/OpenRequestMetadata/index.tsx]: How does the component close? It shouldn't require someone to actually fill it out and hit send in order to close.

File: amundsen_application/static/js/components/TableDetail/OpenRequestMetadata/index.tsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+      this.props.toggleRequest();
+    }
+  }
+
+  render() {
+    return (
+      <a className={ "open-request" }
```
Comment (ttannis) [inline amundsen_application/static/js/components/TableDetail/OpenRequestMetadata/index.tsx]: `className="open-request"` is fine.

Comment (ttannis) [inline amundsen_application/static/js/components/TableDetail/OpenRequestMetadata/index.tsx]: Why would this be an anchor tag? Is it styled in our mock the way that other links are?

File: amundsen_application/static/js/components/TableDetail/OpenRequestMetadata/index.tsx (hunk 2 above)
Comment (ryanlieu) [reply amundsen_application/static/js/components/TableDetail/OpenRequestMetadata/index.tsx]: there's an X button to close

File: amundsen_application/static/js/components/TableDetail/OpenRequestMetadata/index.tsx (hunk 3 above)
Comment (ryanlieu) [reply amundsen_application/static/js/components/TableDetail/OpenRequestMetadata/index.tsx]: keeping styling/behavior consistent with add description button

File: amundsen_application/static/js/components/TableDetail/OpenRequestMetadata/index.tsx (hunk 1 above)
Comment (ryanlieu) [reply amundsen_application/static/js/components/TableDetail/OpenRequestMetadata/index.tsx]: gonna keep this as openrequestmetadata to keep naming consistent with RequestMetadataForm's name

**Comment by ryanlieu** [comment]
@ttannis should be good for review again, lmk if implementation looks good so I can write tests.

File: amundsen_application/static/js/components/TableDetail/OpenRequestMetadata/index.tsx (hunk 1 above)
Comment (ttannis) [reply amundsen_application/static/js/components/TableDetail/OpenRequestMetadata/index.tsx]: You might end up re-considering when we add the other components. More than one type of component will open this modal, so what do they get called? Why would this one be called `OpenRequestMetadata` as opposed to the others? In this sense, you might find it better to name the components even more specifically -- in this case we could  name it according to the feature it serves.

However this is still a nit, let's see what we can do with names down the road.

**Comment by ttannis** [comment]
> @ttannis should be good for review again, lmk if implementation looks good so I can write tests.

Looks good.

**Comment by ryanlieu** [comment]
@ttannis good to look over again with tests, also changed the naming to what you had suggested, definitely makes sense 😄

File: amundsen_application/static/js/components/TableDetail/OpenRequestDescription/styles.scss [hunk 4]
```diff
@@ -0,0 +1,11 @@
+@import 'variables';
+
+.request-description{
+  display: flex;
+  font-size: 16px;
+  text-decoration: none;
+}
+
+.request-description:hover, .request-description:visited, .request-description:active, .request-description:link {
```
Comment (ttannis) [inline amundsen_application/static/js/components/TableDetail/OpenRequestDescription/styles.scss]: formatting nit: Usually each item declaration is on a separate line for easier readability

**Pull request merged**